vector/src/main/java/im/vector/home/CommunicateHomeFragment.kt

```diff
@@ -70,22 +109,21 @@ class CommunicateHomeFragment : BaseCommunicateHomeFragment(), HomeRoomAdapter.O
         val pinMissedNotifications = PreferencesManager.pinMissedNotifications(activity)
         val pinUnreadMessages = PreferencesManager.pinUnreadMessages(activity)
         val notificationComparator = RoomUtils.getNotifCountRoomsComparator(mSession, pinMissedNotifications, pinUnreadMessages)
-        dataUpdateListeners.forEachIndexed { index, listener ->
-            when (index) {
-                ROOM_FRAGMENTS.INVITE.ordinal -> {
-                    listener.onUpdate(mActivity.roomInvitations, notificationComparator)
-                }
-                ROOM_FRAGMENTS.FAVORITE.ordinal -> {
-                    listener.onUpdate(result?.favourites, notificationComparator)
+        calculateRoomPosition()
+        pager.adapter?.notifyDataSetChanged()
+        dataUpdateListeners.forEach { listenerMap ->
+            when (listenerMap.key) {
+                ROOM_FRAGMENTS.INVITE -> {
+                    listenerMap.value?.onUpdate(mActivity.roomInvitations, notificationComparator)
                 }
-                ROOM_FRAGMENTS.NORMAL.ordinal -> {
-                    listener.onUpdate(result?.otherRooms?.plus(result.directChats), notificationComparator)
+                ROOM_FRAGMENTS.FAVORITE -> {
+                    listenerMap.value?.onUpdate(result?.favourites, notificationComparator)
                 }
-                ROOM_FRAGMENTS.LOW_PRIORITY.ordinal -> {
-                    listener.onUpdate(result?.lowPriorities, notificationComparator)
+                ROOM_FRAGMENTS.NORMAL -> {
+                    listenerMap.value?.onUpdate(result?.otherRooms?.plus(result.directChats), notificationComparator)
                 }
-                else -> {
```

Comment:
since its enum, it may not need the else block. If a new key is added, Kotlin usually gives a compile-time error. in this case its showing "redundant else block"